src/test/HST/ApplicationTests.hs

```diff
@@ -0,0 +1,139 @@
+-- | This module contains basic tests for 'haskell-source-transformations'
+module HST.ApplicationTests
+  ( applicationTests
+  )
+where
+
+import           Test.Hspec                     ( Spec
+                                                , Expectation
+                                                , describe
+                                                , it
+                                                , shouldBe
+                                                )
+import           Test.HUnit.Base                ( assertFailure )
+
+import           Control.Monad                  ( void )
+
+import           Language.Haskell.Exts          ( Module
+                                                , ParseResult(..)
+                                                , parseModule
+                                                )
+import           Language.Haskell.Exts.Pretty   ( Pretty
+                                                , prettyPrint
+                                                )
+
+import           HST.Application                ( processModule
+                                                , specialCons
+                                                )
+import           HST.Environment.FreshVars      ( PMState(..)
+                                                , evalPM
+                                                )
+
+-- | Test group for basic unit tests
+applicationTests :: Spec
+applicationTests = describe "Basic unit tests" tests
```

Comment:
Since the module has been renamed, the comment and test description should be updated as well. Furthermore, I'd suggest to use the naming scheme for test function that is used in the [example](https://github.com/FreeProving/guidelines/blob/main/CONTRIBUTING.md#writing-unit-tests) in the guidelines.

```suggestion
-- | Test group for "HST.Application" tests.
testApplication :: Spec
testApplication = describe "HST.Application" $ do
  testProcessModule
```

Reply:
The comment and function name still needs to be updated.